htlcswitch/link.go

```diff
@@ -983,6 +984,16 @@ out:
 	}
 }
 
+// randomFeeUpdateTimeout returns a random timeout between the bounds defined
+// within the link's configuration that will be used to determine when the link
+// should propose an update to its commitment fee rate.
+func (l *channelLink) randomFeeUpdateTimeout() time.Duration {
+	lower := int64(l.cfg.MinFeeUpdateTimeout)
+	upper := int64(l.cfg.MaxFeeUpdateTimeout)
+	rand := prand.Int63n(upper) + lower + 1
```

Comment:
This can cause is to accept (in the worst case), a value of `lower+upper+1`, rather than [lower, upper). Instead we can do something like the following: 
```go
lower := int64(l.cfg.MinFeeUpdateTimeout)
upper := int64(l.cfg.MaxFeeUpdateTimeout)
rand := prand.Int63n(upper)
if rand < lower {
   rand = lower
}
```

Reply:
Fixed.